components/brave_wallet/browser/zcash/zcash_rpc.cc

```diff
@@ -515,4 +533,27 @@ void ZCashRpc::OnGetAddressTxResponse(
   std::move(callback).Run(result.value());
 }
 
+// ZCashDecoder initialization
+void ZCashRpc::InitializeDecoderAndRunCallback(
+    base::OnceCallback<void(bool)> callback) {
+  if (IsConnected()) {
+    std::move(callback).Run(true);
+    return;
+  }
+
+  BraveWalletUtilsService::GetInstance()->CreateZCashDecoder(
+      zcash_decoder_.BindNewEndpointAndPassReceiver(),
+      base::BindOnce(&ZCashRpc::OnDecoderCreated,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void ZCashRpc::OnDecoderCreated(base::OnceCallback<void(bool)> callback) {
+  std::move(callback).Run(IsConnected());
+  zcash_decoder_.reset_on_disconnect();
+}
+
+bool ZCashRpc::IsConnected() {
+  return zcash_decoder_.is_bound();
+}
+
```

Comment:
Why ZcashRpc is doing all of this?
Let's follow pattern of how data_decoder is used:
```
data_decoder::DataDecoder::ParseJsonIsolated(
        fake_response, base::BindOnce(&PhotosService::OnJsonParsed,
                                      weak_factory_.GetWeakPtr(), ""));
```

Reply:
BraveWalletUtilsService could provide various interfaces for different wallet parts, mixing their methods in a single interface would be messy